app/services/analytics.rb

```diff
@@ -95,12 +96,36 @@ def browser_attributes
   end
 
   def session_duration
-    @session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
+    session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
   end
 
   def session_started_at
-    value = @session[:session_started_at]
+    value = session[:session_started_at]
     return value unless value.is_a?(String)
     Time.zone.parse(value)
   end
+
+  def sp_request_attributes
+    resolved_result = resolved_authn_context_result
+    return resolved_result if resolved_result.nil?
```

Comment:
suggestion to make it clearer it's returning nil?
```suggestion
    return if resolved_result.nil?
```